package/MDAnalysis/analysis/dielectric.py

```diff
@@ -0,0 +1,146 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+
+r"""
+Dielectric --- :mod:`MDAnalysis.analysis.dielectric`
+===========================================================
+
+:Authors: Mattia Felice Palermo, Philip Loche
+:Year: 2018
+:Copyright: GNU Public License v2
+
+A tool to compute the average dipole moment and the static dielectric constant
+
+.. math::
+
+   \varepsilon = 1 + \frac{\langle M^2 \rangle - \langle M \rangle^2}
+                            {3 \varepsilon_ 0 V k_B T}
+
+for a system simulated in tin foil boundary conditions. For a distance
+
+References
+----------
+Neumann, Martin.
+Dipole Moment Fluctuation Formulas in Computer Simulations of Polar Systems. 
+Molecular Physics 50, no. 4 (November 1983), 841–858, doi:10.1080/00268978300102721
+"""
+
+from __future__ import division
+
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from MDAnalysis.units import constants, convert
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.mdamath import make_whole
+
+class DielectricConstant(AnalysisBase):
+    r"""Dielectric constant
+
+    Parameters
+    ----------
+    selection : AtomGroup
+          any atomgroup
+    start : int
+          The frame to start at [0]
+    stop : int
+          The frame to end at [-1]
+    step : int
+          The step size through the trajectory in frames [0]
+    temperature : float
+        Temperature (Kelvin) at which the system has been simulated [298.15]
+    verbose : bool (optional)
+          Show detailed progress of the calculation if set to ``True``; the
+          default is ``False``.
+        
+    Attributes
+    ----------
+    results : dict
```

Comment:
I would like to see the units each value is in.

Reply:
done